Add EventTriggersFunctionOperator to Kafka Integrations page (#1657)
astronomer/docs: learn/airflow-kafka.md

- `poll_interval`: The amonut of time in seconds that the task should wait in the deferred state.
- `xcom_push_key`: The key under which to save the returned data to XCom.

### EventTriggersFunctionOperator

> The EventTriggerFunctionOperator is a [deferrable operator](https://docs.astronomer.io/learn/deferrable-operators) that will wait for a specific message to be published to one or more Kafka topics, similar to the AwaitKafkaMessageOperator. After consuming a message that fulfills the criteria defined in the `apply_function` the EventTriggerFunctionOperator will not complete its task like the AwaitKafkaMessageOperator, instead it will defer itself again and continue listening until the task is stopped externally for example by a timeout value of the DAG itself. Without an external termination, this task will keep deferring itself indefinitely as long as the DAG is running. View [an example DAG using this operator](https://github.com/astronomer/airflow-provider-kafka/blob/main/example_dags/listener_dag_function.py) in the provider repository.

Reviewer: ```suggestion
The EventTriggerFunctionOperator is a [deferrable operator](https://docs.astronomer.io/learn/deferrable-operators) that waits for a specific message to be published to one or more Kafka topics, similar to the AwaitKafkaMessageOperator. Unlike the AwaitKafkaMessageOperator, after consuming a message that fulfills the criteria defined in the `apply_function` parameter, the EventTriggerFunctionOperator will continue listening until the task is stopped by an external criteria such as a timeout of the DAG itself. Without external termination, this task will stay in a deferred state indefinitely as long as the DAG is running. You can view [an example DAG using this operator](https://github.com/astronomer/airflow-provider-kafka/blob/main/example_dags/listener_dag_function.py) in the provider repository.
```

> 

Reviewer: You can define the following parameters for this operator:

- `topics`: A list of topics or regex patterns to read from.

> - `apply_function`: A Python function that is applied to all messages that are read. If the function returns any data the `event_triggered_function` will run.

Reviewer: ```suggestion
- `apply_function`: A Python function that is applied to all messages that are read. If the function returns any data, the `event_triggered_function` will run.
```

> - `event_triggered_function`: A Python function that will be triggered every time the operator consumes a message that cause the `apply_function` to return any data. The value returned by the apply_function is passed to the function provided to `event_triggered_function` as the first positional parameter.

Reviewer: ```suggestion
- `event_triggered_function`: A Python function that will be triggered every time the operator consumes a message that causes the `apply_function` to return any data. The value returned by the `apply_function` is passed to the function provided to `event_triggered_function` as the first positional parameter.
```

- `poll_interval`: The amount of time in seconds that the task should wait in the deferred state.
- `xcom_push_key`: The key under which to save the returned data to XCom.

### EventTriggersFunctionOperator

> The EventTriggersFunctionOperator is a [deferrable operator](https://docs.astronomer.io/learn/deferrable-operators) that waits for a specific message to be published to one or more Kafka topics, similar to the AwaitKafkaMessageOperator. Unlike the AwaitKafkaMessageOperator, the EventTriggersFunctionOperator will continue listening until the task is stopped by an external criteria such as a timeout of the DAG itself. If this external criteria isn't met, this task will stay in a deferred state indefinitely as long as the DAG is running. You can view an example DAG using this operator in the [`astronomer-providers` repository](https://github.com/astronomer/airflow-provider-kafka/blob/main/example_dags/listener_dag_function.py).

Author: ```suggestion
The EventTriggersFunctionOperator is a [deferrable operator](https://docs.astronomer.io/learn/deferrable-operators) that waits for a specific message to be published to one or more Kafka topics, similar to the AwaitKafkaMessageOperator. Unlike the AwaitKafkaMessageOperator, the EventTriggersFunctionOperator will continue listening until the task is stopped by an external criterion such as a timeout of the DAG itself. If this external criteria isn't met, this task will stay in a deferred state indefinitely as long as the DAG is running. You can view an example DAG using this operator in the [`astronomer-providers` repository](https://github.com/astronomer/airflow-provider-kafka/blob/main/example_dags/listener_dag_function.py).
```

> The EventTriggerFunctionOperator is a [deferrable operator](https://docs.astronomer.io/learn/deferrable-operators) that waits for a specific message to be published to one or more Kafka topics, similar to the AwaitKafkaMessageOperator. Unlike the AwaitKafkaMessageOperator, after consuming a message that fulfills the criteria defined in the `apply_function` parameter, the EventTriggerFunctionOperator will continue listening until the task is stopped by an external criteria such as a timeout of the DAG itself. Without external termination, this task will stay in a deferred state indefinitely as long as the DAG is running. You can view [an example DAG using this operator](https://github.com/astronomer/airflow-provider-kafka/blob/main/example_dags/listener_dag_function.py) in the provider repository.

Reviewer 2: ```suggestion
The EventTriggersFunctionOperator is a [deferrable operator](https://docs.astronomer.io/learn/deferrable-operators) that waits for a specific message to be published to one or more Kafka topics, similar to the AwaitKafkaMessageOperator. Unlike the AwaitKafkaMessageOperator, the EventTriggersFunctionOperator will continue listening until the task is stopped by an external criteria such as a timeout of the DAG itself. If this external criteria isn't met, this task will stay in a deferred state indefinitely as long as the DAG is running. You can view an example DAG using this operator in the [`astronomer-providers` repository](https://github.com/astronomer/airflow-provider-kafka/blob/main/example_dags/listener_dag_function.py).
```

You can define the following parameters for this operator:

- `topics`: A list of topics or regex patterns to read from.
- `apply_function`: A Python function that is applied to all messages that are read. If the function returns any data, the `event_triggered_function` will run.

> - `event_triggered_function`: A Python function that will be triggered every time the operator consumes a message that causes the `apply_function` to return any data. The value returned by the `apply_function` is passed to the function provided to `event_triggered_function` as the first positional parameter.

Reviewer 2: ```suggestion
- `event_triggered_function`: A Python function that runs every time the operator consumes a message that causes the `apply_function` to return any data. The value returned by the `apply_function` is passed to the function provided to `event_triggered_function` as the first positional parameter.
```

- `apply_function_args`: Positional arguments for the `apply_function`.
- `apply_function_kwargs`: Keyword arguments for the `apply_function`.

> - `kafka_config`: The configuration for the Kafka client, including the connection information. For a full list of parameters please refer to the [librdkafka GitHub repository](https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md).

Reviewer 2: ```suggestion
- `kafka_config`: The configuration for the Kafka client, including the connection information. For a full list of parameters, see the [librdkafka GitHub repository](https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md).
```

- `poll_timeout`: The amount of time in seconds that the task should wait for a message in its active state.

> - `poll_interval`: The amonut of time in seconds that the task should wait in the deferred state.

Reviewer 2: ```suggestion
- `poll_interval`: The amount of time in seconds that the task should wait in the deferred state.
```

Author: Second time today I made that typo. Maybe a subconscious wish for donuts? 😅
